Cython/Compiler/Optimize.py

```diff
@@ -1440,6 +1440,11 @@ def visit_PrimaryCmpNode(self, node):
             # note: lhs may have side effects
             return node
 
+        if any([arg.is_starred for arg in args]):
+            # note: Ignore optimisations if there is at least 
+            # one starred expression in the args
```

Comment:
This is a redundant comment. The code does exactly what the comment says. Let's make it a bit more interesting by giving an actual reason.
```suggestion
            # Starred arguments do not directly translate to comparisons or "in" tests.
```